analytics/apigee/lib/apigeeanalytics.js

```diff
@@ -120,8 +120,10 @@ ApigeeAnalyticsSpi.prototype.makeRecord = function(req, resp, cb) {
 
   var self = this;
   onFinished(resp, function() {
+    var now = Date.now();
     record.response_status_code      = resp.statusCode;
-    record.client_sent_end_timestamp = Date.now();
+    record.client_sent_start_timestamp = now;
```

Comment:
I think the field names are really confusing here. It threw me off as well.

client_received_start_timestamp and client_received_end_timestamp are timestamps when Server got the first bit from client and when server got last bit from client respectively.

client_sent_start_timestamp and client_sent_end_timestamp are timestamps when Server sent first response bit to client and sent last bit to client respectively.

Delta will be noticeable when request or response payload is relatively big.